### PR TITLE
Web security fundamentals

### PR DESCRIPTION
**Pull request by socjopata**

### CODE REVIEW

File: content/web_development/skills/web-security-fundamentals.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ ##### 📗 The web framework of my choice is awesome, because it acts as my safety net:
+ - Depending on your future development path please pick either
+     - [Backend](https://guides.rubyonrails.org/security.html)
+     - Frontend
+         - [1](https://snyk.io/blog/10-react-security-best-practices/)
+         - [2](https://relevant.software/blog/react-js-security-guide/)
+ - and tell me about one cool thing you've learnt there.
+ 
+ ##### 📗 Protecting yourself and others sometimes means that you have to pose a threat by yourself.
+ 📗 Imagine that your friend recommended you an article, that is behind a paywall. You're broke and can't afford to honestly pay the author, but the information in that article are essential for your survival. What could you possibly do to obtain such information (and why it's possible? Bonus points for covering all the JS, HTML and CSS)
```
Comment (PoslinskiNet) [inline content/web_development/skills/web-security-fundamentals.md:67]: It is more an abstract kata or do you have something practical in mind?

Comment (socjopata) [reply content/web_development/skills/web-security-fundamentals.md:67]: It's abstract and an opportunity to have a discussion with the mentee. Imho a lot of problems with our juniors participating in critical security failure incidents come from the lack of perspective and abstractions when it comes to the solutions they do author. Breaking the walls built by Javascript, HTML and CSS should give them some second thoughts in the future (I hope)

**Comment by PoslinskiNet** [comment]
Any thoughts @stevo @qbart @mikoscz?

**Comment by mikoscz** [comment]
What is the context behind that? It would be a bubble/prerequisit?

File: content/web_development/skills/web-security-fundamentals.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- 📗 [Did you know that you can actually browse cookies?](https://www.troyhunt.com/how-to-build-and-how-not-to-build/)
+- 📗 [A few words as to where put your trust](https://www.codebyamir.com/blog/never-trust-data-from-the-browser)
+- 📗 [Become a hacker yourself or just stop being naive with your solutions. How to get get behind a paywall](https://medium.datadriveninvestor.com/how-to-bypass-any-paywall-for-free-df87832cbff7)
+
+
+### 🎤 Interview
+
+- Can you provide some examples of a sensitive data found in a web or mobile application? What makes such data "sensitive"?
+- What's the original, intended use of OAuth? What kind of challenges OAuth authors tried to solve in the first place?
+- Please provide a list different approaches to client side data storage. Which approaches are safe for storing sensitive data like passwords?
```
[the remark this answers is not in the document]
Comment (socjopata) [reply content/web_development/skills/web-security-fundamentals.md:46]: localStorage, sessionStorage (both unsafe, one expiring) and Cookies

[the remark this answers is not in the document]
Comment (qbart) [reply content/web_development/skills/web-security-fundamentals.md:46]: i get it but can we rewrite question so it does not suggest keeping passwords client side?

File: content/web_development/skills/web-security-fundamentals.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+**Crash course to everything you should need to know about web security**
+
+---
+
+### 🎓 Learn (agnostic part)
+
+- 📗 [OAuth in 6 minutes](https://www.youtube.com/watch?v=KT8ybowdyr0),
+- 📗 [Three ways of storing data on the client side (for starters)](https://krishankantsinghal.medium.com/local-storage-vs-session-storage-vs-cookie-22655ff75a8)
+- 📗 [HTTPS, TLS, SSL, ETC, PDK... ;-)](https://kinsta.com/knowledgebase/tls-vs-ssl)
+- 📗 [Oh, there is a padlock icon. I'm safe... NOT!](https://niebezpiecznik.pl/post/przestan-sprawdzac-czy-strony-maja-klodki/)
```
[the remark this answers is not in the document]
Comment (qbart) [reply content/web_development/skills/web-security-fundamentals.md:36]: i've noticed one more thing:
> HTTPS, TLS, SSL, ETC, PDK... ;-)
let's remove all the jokes ;)

**Comment by PoslinskiNet** [comment]
@qbart WDYT?

**Comment by qbart** [comment]
@socjopata @PoslinskiNet It's a relatively small piece, we can adjust things later, I am ok with merge, @mikoscz ?

**Comment by PoslinskiNet** [comment]
No objections, I guess we can merge it and adjust later if needed.

**Pull request merged**